### ISSUE TITLE
numberFormat expression rounding

### ISSUE THREAD

**Issue by abouroubi** [mapbox/mapbox-maps-ios#2024]
## Environment

- Xcode version: 14.3.1
- iOS version: 16.4
- Devices affected: all
- Maps SDK Version: 10.16.0

## Observed behavior and steps to reproduce

I'm using a symbol layer to show numbers on a map.
I'm trying to use the numberFormat expression, to add currency to the number. 
But I don't know why, the number format is rounding the value I'm sending to it.

Here is my code:
```swift
mapView.mapboxMap.onNext(.styleLoaded) { _ in
    // Create a source
    let source = GeoJSONSource()
    source.data = .feature(Feature(geometry: .point(Point(CLLocationCoordinate2D(latitude: 40.73, longitude: -74.0)))))
    
    try! mapView.mapboxMap.style.addSource(source, id: "point-source")

 let formatOptions = NumberFormatOptions(locale: "en-US", currency: "USD", minFractionDigits: 2, maxFractionDigits: 2)
        
let textExp = Exp(.numberFormat) {
    2.6
    formatOptions
}

var layer = SymbolLayer(id: "text-layer")
layer.source = "point-source"
layer.textField = .expression(textExp)
try! mapView.mapboxMap.style.addLayer(layer)
```
And the text shown is $3

## Expected behavior

The text should be $2.60 

## Notes / preliminary analysis

Number format is rounding the variable sent to it.

**Comment by OdNairy** [comment]
Hi @abouroubi 
Thank you for a report with a working example. I can confirm that the issue is reproducible.
I've opened a ticket in internal system MAPSNAT-1425

**Comment by abouroubi** [comment]
> Hi @abouroubi Thank you for a report with a working example. I can confirm that the issue is reproducible. I've opened a ticket in internal system [MAPSNAT-1425](https://mapbox.atlassian.net/browse/MAPSNAT-1425)

Thanks, do you have an ETA for the fix ? This is the last thing keeping us from going live.

**Comment by OdNairy** [comment]
Hey @abouroubi 
I have good news - the rendering team has already addressed the issue and is going to include the fix in the upcoming 10.16.1 release this week.
